feat(calendar): Readded multiple groups and added blocked groups

lwcorp opened the pull request:
# Description

1. Undid the removal of the support for multiple groups (why did you allow this person's PR just so they fix formatting?)
1. Made the multiple groups and get all members functions to use each other to avoid duplicated code
1. Added support for avoiding blocked groups (e.g. external ones)
1. Simplified both functions' code
1. Fixed spelling while at it (hope it's okay to avoid an extra PR just to put `fix` in the title instead of `feat`)

**Update** - removed as newer updates stopped checking summaries:
<del>6. Fixed to prevent script crash when any event has no summary</del>

## Is it been tested?
- [X] Development testing done

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have performed a peer-reviewed with team member(s)
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules

lwcorp commented:
Please note the failed tests failed Google's own original code, not my additions.

baveer reviewed (approved):
The code works great for single group, nested groups and multiple groups (which was broken by someone else). I have no idea why these tests failed, several calendars are up and running in my environment.

lwcorp committed:
feat(calendar): Readded multiple groups and added blocked groups

1. Undid the removal of the support for multiple groups (why did you allow this person's PR just so they fix formatting?)
2. Made the multiple groups and get all members functions to use each other to avoid duplicated code
3. Added support for avoiding blocked groups (e.g. external ones)
4. Simplified both functions' code
5. Fixed spelling while at it (hope it's okay to avoid an extra PR just to put `fix` in the title instead of `feat`)

lwcorp commented:
Refactored code to match newer merged PRs.
Please consider approving as it's shame to lose a the vital multiple groups' feature which you've accidentally removed.
